emulator.c

```diff
@@ -19,10 +19,16 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
             cpu->d += 1;
           }
 
-        cpu->pc += 1;
         break;
       }
+    default:
+      {
+        fprintf(stderr, "Error: opcode 0x%02x not found\n", opcode);
+        return -1;
+      }
     }
+  cpu->pc += 1;
```

Comment:
Is there ever a situation where we wouldn't increment the counter in the switch?

Reply:
Great call out - I think JMP, CALL, RET may not actually increment the counter. They would all pop or push addresses onto the stack and then move the stack pointer to another place in memory.

Reply:
So as I'm thinking about this, does that mean changes to pc need to be moved back into the case statements?